controls/radmap/how-to/howto-wraparound-virtualization-visualizationlayer.md

```diff
@@ -0,0 +1,383 @@
+
+![Wraparound with Virtualization in VisualizationLayer](images/RadMap_Wraparound_Virtualization_Visualizationlayer.PNG)	
+
+In order to enable the UI Virtualization feature, you should configure the ZoomLevelGridList object and you should specify the VirtualizationSource property of the [Visualization Layer]({%slug radmap-features-virtualization-layer%}). The ZoomLevelGridList property allows dividing the map into regions. The data provided to the VisualizationLayer should be wrapped in a class that implements the __IMapItemsVirtualizationSource__ interface. This interface contains a __MapItemsRequest()__ method, which is used by the __VisualizationLayer__ to request new data whenever the ZoomLevel or Center properties of the RadMap control are changed. The event arguments are of type __MapItemsRequestEventArgs__ which give you the currently traversed portion of the map. This method generally serves as an event handler and you have to calculate and decide which objects to be displayed in the given portion. In WrapAround mode, this portion will have Longitude value exceeding the valid geographical boundaries (-180; 180). The RadMap control provides public methods which you can use to normalize such locations. You can take a look a the [Location Conversion]({%slug radmap-features-wraparound%}) section in the WrapAround help article.
+
+In the next part of this article, we will display points of interest (POIs) over the cities (Berlin, London, Amsterdam, Madrid, Barcelona, Rome) in the map in MVVM scenario and set the virtualization when the wraparound mode of the map is on. In __Example 1__ we will specify the business object which will hold the Longitude, Latitude values, the name of the cities and their location.
+
+#### __[C#] Example 1:  Business object__
+{{region c#-radmap-radmap-howto-wraparound-virtualization-visualizationlayer_0}}
+	public class CityLocation : ViewModelBase
+    {
+        private double _lattitude;
+        private double _longitude;
+        private string _cityName;
+        public CityLocation(double latitude, double longitude, string cityName)
+        {
+            this.Latitude = latitude;
+            this.Longitude = longitude;
+            this.CityName = cityName;
+        }
+
+        public double Latitude
+        {
+            get
+            {
+                return this._lattitude;
+            }
+            set
+            {
+                Location old = this.Location;
+                this._lattitude = value;
+                this.OnPropertyChanged("Latitude");
+                this.OnPropertyChanged("Location");
+            }
+        }
+
+        public double Longitude
+        {
+            get
+            {
+                return this._longitude;
+            }
+            set
+            {
+                Location old = this.Location;
+
+                this._longitude = value;
+                this.OnPropertyChanged("Longitude");
```

Comment:
I don't think we need to raise PropertyChanged for the Longitude and Latitude properties. Because their are not bound in the view. Instead we can raise it only for the Location property. However, I am not sure, so test this.